src/scripts/main.js

```diff
@@ -1 +1,72 @@
 'use strict';
+
+// write your code here
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', (ev) => {
+    if (ev.button === 0) {
+      resolve('First promise was resolved');
+    }
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', (ev) => {
+    if (ev.button === 0 || ev.button === 2) {
+      resolve('Second promise was resolved');
+    }
+  });
+});
+
+let leftCnt = 0;
+let rightCnt = 0;
+const thirdPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', (ev) => {
+    // ev.preventDefault();
+
+    if (ev.button === 0) {
+      leftCnt++;
+    }
+
+    if (ev.button === 2) {
+      rightCnt++;
+    }
```

Comment:
From what I checked, something doesn't really work with the right click here, same with the second promise. This could be perhaps dealt with by replacing 'click' with 'mouseup' or 'mousedown'